server/app/controllers/faq_controller.rb

```diff
@@ -0,0 +1,47 @@
+class FaqController < ApplicationController
+    before_action :is_admin, except: [:show, :index]
+
+    def index
+        @faqs = Faq.all
+        render json: { faqs: faqs }
+    end
+
+    def show
+        @faq = Faq.find(params[:id])
+        render json: { faq: faq }
+    end
+
+    def create
+            faq = Faq.create(faq_params)
+            render json: { faq: faq }
+    end
+
+    def destroy
+            @faq = Faq.find(params[:id])
+            @faq.destroy
+            render json: { status: 200, 
+                message: 'Faq deleted successfuly'
+            }
+
+    end
+
+    def update
+        @faq = Faq.find(params[:id])
+    
+        if @faq.update(faq_params)
+            render json: { status: 200, 
+                message: 'Faq updated successfuly'
+            }
+        else
+          render json: {
+            status: 400,
+            errors: ["Could not edit faq"],
+            }
+        end
+    end
+
+    def faq_params
+        params.require(:faq).permit(:question, :answer)
+    end
+end
```

Comment:
Tests for routes?

Reply:
In this case, it looks like doing a request spec should take care of the routes automatically. Agree?

Reply:
I think we're supposed to have rspec and cucumber tests for everything, if possible

Reply:
I am really bad at cucumber tests. My thinking is either one is fine. I believe I heard that in one of the lectures

Reply:
No the model testing isnt the same as route testing

Reply:
For example your rspec test doesnt touch a single line of code in the controller cose

Reply:
I'm aware. I've been trying to write tests for routes but it keeps giving "missing controller: FaqController" but the controller does exist